YoutubeDownloader/Views/Dialogs/AuthSetupView.xaml.cs

```diff
@@ -30,53 +35,48 @@ private void LogoutHyperlink_OnClick(object sender, RoutedEventArgs args)
-        WebBrowser.CoreWebView2.Settings.IsStatusBarEnabled = false;
-        WebBrowser.CoreWebView2.Settings.IsSwipeNavigationEnabled = false;
-    }
+        var platformWebView = (
+            (Avalonia.WebView.Windows.Core.WebView2Core)WebBrowser.PlatformWebView!
+        );
+        _coreWebView2 = platformWebView!.CoreWebView2!;
 
-    private void WebBrowser_OnNavigationStarting(
-        object? sender,
-        CoreWebView2NavigationStartingEventArgs args
-    )
-    {
-        // Reset existing browser cookies if the user is attempting to log in (again)
-        if (string.Equals(args.Uri, LoginPageUrl, StringComparison.OrdinalIgnoreCase))
-            WebBrowser.CoreWebView2.CookieManager.DeleteAllCookies();
+        _coreWebView2.Settings.AreDefaultContextMenusEnabled = false;
+        _coreWebView2.Settings.AreDevToolsEnabled = false;
+        _coreWebView2.Settings.IsGeneralAutofillEnabled = false;
+        _coreWebView2.Settings.IsPasswordAutosaveEnabled = false;
+        _coreWebView2.Settings.IsStatusBarEnabled = false;
+        _coreWebView2.Settings.IsSwipeNavigationEnabled = false;
     }
 
-    private async void WebBrowser_OnNavigationCompleted(
+    private async void WebBrowser_OnNavigationStarting(
         object? sender,
-        CoreWebView2NavigationCompletedEventArgs args
+        WebViewUrlLoadingEventArg args
     )
     {
-        var url = WebBrowser.Source.AbsoluteUri.TrimEnd('/');
+        // Reset existing browser cookies if the user is attempting to log in (again)
+        if (string.Equals(args.Url?.AbsoluteUri, LoginPageUrl, StringComparison.OrdinalIgnoreCase))
+            _coreWebView2!.CookieManager.DeleteAllCookies();
 
-        // Navigated to the home page (presumably after a successful login)
-        if (string.Equals(url, HomePageUrl, StringComparison.OrdinalIgnoreCase))
+        if (args.Url!.AbsoluteUri.StartsWith(HomePageUrl, StringComparison.OrdinalIgnoreCase))
         {
             // Extract the cookies that the browser received after logging in
-            var cookies = await WebBrowser.CoreWebView2.CookieManager.GetCookiesAsync(url);
+            var cookies = await _coreWebView2!.CookieManager.GetCookiesAsync(
+                args.Url!.AbsoluteUri!
+            );
             ViewModel.Cookies = cookies.Select(c => c.ToSystemNetCookie()).ToArray();
         }
     }
 
-    private async void WebBrowser_OnUnloaded(object sender, RoutedEventArgs args)
+    private void WebBrowser_OnUnloaded(object? sender, RoutedEventArgs e)
     {
         // This will most likely not work because WebView2 would still be running at this point,
         // and there doesn't seem to be any way to shut it down using the .NET API.
-        if (WebBrowser.CoreWebView2?.Profile is not null)
-            await WebBrowser.CoreWebView2.Profile.ClearBrowsingDataAsync();
+        // if (_coreWebView2?.Profile is not null)
+        //     await _coreWebView2.Profile.ClearBrowsingDataAsync();
     }
```

Comment:
Are these methods not accessible or why are they commented out?

Reply:
the webview is already disposed at that point so it throws an exception:
```
System.InvalidOperationException: CoreWebView2 members cannot be accessed after the WebView2 control is disposed.
```

So I guess the handler can just be removed.